ibm/service/power/data_source_ibm_pi_volume.go

```diff
@@ -144,14 +165,20 @@ func dataSourceIBMPIVolumeRead(ctx context.Context, d *schema.ResourceData, meta
 	d.Set(Attr_AuxiliaryVolumeName, volumedata.AuxVolumeName)
 	d.Set(Attr_Bootable, volumedata.Bootable)
 	d.Set(Attr_ConsistencyGroupName, volumedata.ConsistencyGroupName)
+	d.Set(Attr_CreationDate, volumedata.CreationDate.String())
 	d.Set(Attr_DiskType, volumedata.DiskType)
+	if volumedata.FreezeTime != nil {
+		d.Set(Attr_FreezeTime, volumedata.FreezeTime.String())
+	}
 	d.Set(Attr_GroupID, volumedata.GroupID)
 	d.Set(Attr_IOThrottleRate, volumedata.IoThrottleRate)
+	d.Set(Attr_LastUpdateDate, volumedata.LastUpdateDate.String())
 	d.Set(Attr_MasterVolumeName, volumedata.MasterVolumeName)
 	d.Set(Attr_MirroringState, volumedata.MirroringState)
 	d.Set(Attr_PrimaryRole, volumedata.PrimaryRole)
 	d.Set(Attr_ReplicationEnabled, volumedata.ReplicationEnabled)
 	d.Set(Attr_ReplicationType, volumedata.ReplicationType)
+	d.Set(Attr_ReplicationSites, volumedata.ReplicationSites)
```

Comment:
Could you add check Len()> 0 before setting it?

Reply:
Added check to instance volumes and volume data sources in next commit.